scripts/o3de/o3de/project_properties.py

```diff
@@ -81,47 +79,89 @@ def edit_project_props(proj_path: pathlib.Path = None,
-                       is_optional: bool = False
+                       new_compatible_engines: str or list = None,
+                       delete_compatible_engines: str or list = None,
+                       replace_compatible_engines: str or list = None,
+                       new_version: str = None,
+                       is_optional_gem: bool = False,
+                       new_engine_api_dependencies: list or str = None,
+                       delete_engine_api_dependencies: list or str = None,
+                       replace_engine_api_dependencies: list or str = None
                        ) -> int:
+    """
+    Edits and modifies the project properties for the project located at 'proj_path' or with the name 'proj_name'.
+    :param proj_path: The path to the project folder
+    :param proj_name: The name of the project 
+    :param new_name: The new name for the project
+    :param new_id: The new ID for the project
+    :param new_origin: The new origin for the project
+    :param new_display: The new display name for the project
+    :param new_summary: The new gem summary text
+    :param new_icon: The new path to the gem's icon file 
+    :param new_tags: New tags to add to 'user_tags'
+    :param delete_tags: Tags to remove from 'user_tags'
+    :param replace_tags: Tags to replace 'user_tags' with
+    :param new_gem_names: Any gem names to be added to the list
+    :param delete_gem_names: Any gem names to be removed from the list
+    :param replace_gem_names: A list of gem names that will completely replace the current list
+    :param new_engine_name: The new engine name this project is registered with
+    :param new_compatible_engines: Compatible engine version specifiers to add e.g. o3de==1.2.3
+    :param delete_compatible_engines: Engine version specifiers to remove from 'compatible_engines'
+    :param replace_compatible_engines: Engine version specifiers to replace everything in'compatible_engines' 
+    :param new_version: The new project version e.g. 1.2.3
+    :param is_optional_gem: Only applies to new_gem_names, when true will add an 'optional' property to the gem(s)
+    :param new_engine_api_dependencies: Engine API version specifiers to add e.g. launcher==1.2.3
+    :param remove_engine_api_dependencies: Version specifiers to remove from 'engine_api_dependencies'
+    :param replace_engine_api_dependencies: Version specifiers to replace 'engine_api_dependencies' with
+    """
     proj_json = get_project_props(proj_name, proj_path)
 
     if not proj_json:
         return 1
-    if new_name:
+    if isinstance(new_name, str):
         if not utils.validate_identifier(new_name):
             logger.error(f'Project name must be fewer than 64 characters, contain only alphanumeric, "_" or "-" characters, and start with a letter.  {new_name}')
             return 1
         proj_json['project_name'] = new_name
-    if new_id:
+    if isinstance(new_id, str):
```

Comment:
```new_id``` (and other arguments) are tagged as ```str``` in the argument list. Just wondering why we are checking the instancetype here instead of just relying on it not being Null or empty?

Reply:
we want to allow the user to set the id field to an empty string, and this is basically following the pattern we use in other python files allowing that.